Documentation/ceph-csi-drivers.md

```diff
@@ -124,5 +124,77 @@ when pod get spawned and destroyed at time of pod delete.
 Refer to [ephemeral-doc](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes) for more info.
 Also, See the example manifests for an [RBD ephemeral volume](https://github.com/rook/rook/tree/{{ branchName }}/deploy/examples/csi/rbd/pod-ephemeral.yaml) and a [CephFS ephemeral volume](https://github.com/rook/rook/tree/{{ branchName }}/deploy/examples/csi/cephfs/pod-ephemeral.yaml).
 
+## CSI-Addons Controller
+
+The CSI-Addons Controller handles the requests from users to initiate an operation. Users create a CR that the controller inspects, and forwards a request to one or more CSI-Addons side-cars for execution.
+By listing the CSIAddonsNode CRs, the CSI-Addons Controller knows how to connect to the side-cars. By checking the supported capabilities of the side-cars, it can decide where to execute operations that the user requested.
+
+### Installation by operator
+
+A CSI-Addons bundle can be used to install the CSI-Addons Controller with the following steps:
+
+```bash
+$ kubectl create namespace storage-csi-addons
+$ operator-sdk run bundle -n storage-csi-addons quay.io/csiaddons/k8s-bundle:latest
+```
+
+### Enable the CSI-Addons Sidecar
+
+To use the features provided by the CSI-Addons, the `csi-addons`
+containers need to be deployed in the RBD provisioner and nodeplugin pods,
+which are not enabled by default.
+
+Execute the following command in the cluster to enable the CSI-Addons
+sidecar:
+
+* Edit the `rook-ceph-operator-config` configmap and add the
+ following configurations
+
+```bash
+kubectl edit cm rook-ceph-operator-config -n rook-ceph
+```
+
+* Add the following properties if not present:
+
+```yaml
+data:
+  CSI_ENABLE_CSIADDONS: "true"
+```
+
+* After updating the configmap with those settings, a new sidecar
+ should now start automatically in the CSI provisioner and nodeplugin pod.
+
+NOTE: Make sure the version of ceph-csi used is v3.5.0+
+
+### Reclaim Space Operation
+
+If a user wants to initiate a Reclaim Space operation, he will create a ReclaimSpaceJob
+CR that includes the reclaim space as an operation.
+
+```yaml
+apiVersion: csiaddons.openshift.io/v1alpha1
+kind: ReclaimSpaceJob
```

Comment:
cronjob or annotation is most useful for users can be documented that example?

Reply:
done